gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/processor/transaction/TransactionResponseProcessor.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.handlers.api.processor.transaction;
+
+import io.gravitee.gateway.api.ExecutionContext;
+import io.gravitee.gateway.api.Response;
+import io.gravitee.gateway.api.http.HttpHeaders;
+import io.gravitee.gateway.core.processor.AbstractProcessor;
+
+/**
+ * A {@link Response} processor used to set the transaction ID in the response headers.
+ */
+public class TransactionResponseProcessor extends AbstractProcessor<ExecutionContext> {
+
+    private final TransactionResponseProcessorConfiguration configuration;
+
+    public TransactionResponseProcessor(TransactionResponseProcessorConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public void handle(final ExecutionContext context) {
+        setHeaderAccordingToBackendOverrideMode(
+            context.request().headers(),
+            context.response().headers(),
+            configuration.transactionHeader,
+            configuration.transactionHeaderHeaderOverrideMode
+        );
+
+        setHeaderAccordingToBackendOverrideMode(
+            context.request().headers(),
+            context.response().headers(),
+            configuration.requestHeader,
+            configuration.requestHeaderHeaderOverrideMode
+        );
```

Comment:
Why is it done twice?

Reply:
One for the request id header and the other for the transaction header, as the conf can be different for both

Reply:
I missed the diff between requestId and transactionId 👍